fix: spawn child process with process.env in macOS arm64 (#25753)
cypress-io/cypress: cli/CHANGELOG.md

- Fixed an issue with the Cloud project selection modal not showing the correct prompts. Fixes [#25520](https://github.com/cypress-io/cypress/issues/25520).
- Fixed an issue in middleware where error-handling code could itself generate an error and fail to report the original issue. Fixes [#22825](https://github.com/cypress-io/cypress/issues/22825).

> - Fixed an issue where custom browsers may not be found in macOS arm64 architectures that rely on process environment variables. Fixed in [#25753](https://github.com/cypress-io/cypress/pull/25753).

Reviewer: ```suggestion
- Fixed an issue where custom browsers that relied on process environment variables were not found on macOS arm64 architectures. Fixed in [#25753](https://github.com/cypress-io/cypress/pull/25753).
```
